src/pytest_nm_releng/plugin.py

```diff
@@ -15,18 +15,45 @@
 
 import os
 from datetime import datetime, timezone
+from enum import Enum
 from pathlib import Path
 
+from uuid_utils import uuid4, uuid7
 
-def get_utc_timestamp() -> float:
-    return datetime.now(timezone.utc).timestamp()
+
+class SuffixType(Enum):
+    TIMESTAMP = "timestamp"
+    UUID4 = "uuid4"
+    UUID7 = "uuid7"
+
+
+def get_utc_timestamp() -> str:
+    return str(datetime.now(timezone.utc).timestamp())
+
+
+def generate_suffix(suffix_type: SuffixType) -> str:
+    suffix_map = {
+        SuffixType.TIMESTAMP: get_utc_timestamp,
+        SuffixType.UUID4: lambda: str(uuid4()),
+        SuffixType.UUID7: lambda: str(uuid7()),
+    }
+    return suffix_map.get(suffix_type, get_utc_timestamp)()
 
 
 def generate_junit_flags() -> list[str]:
     if not (junitxml_base_dir := os.getenv("NMRE_JUNIT_BASE")):
         return []
 
-    junitxml_file = Path(junitxml_base_dir) / f"{get_utc_timestamp()}.xml"
+    junit_suffix_type = os.getenv("NMRE_JUNIT_SUFFIX_TYPE")
```

Comment:
why not just set the default `SuffixType.TIMESTAMP` in this method call?

Reply:
Good question, I’ll take a closer look at that.

Reply:
I have updated this area to set the default to be a little extra explicit (and defined the default separately to make any changes to it cleaner), but this actually doesn't reduce any logic below because having the default only helps if the environment variable is unset, not if it is e.g., defined but empty or set to an invalid value.

I suppose there is an argument to be made about invalid values causing errors, but here I was leaning toward just replacing with the default value (maybe emit a warning?). If you feel differently, let me know (also if you consider an empty value to be in the same vein as other invalid values).

Reply:
i'm good.  thanks for the explaination